packages/llm/distributed.py

```
@@ -0,0 +1,338 @@
+        try:
+            import redis.asyncio as aioredis
+        except ImportError:
+            log.info("llm.distributed: redis package absent — pacing locally")
+            return None
+        try:
+            client = aioredis.from_url(self._redis_url, decode_responses=True)
+            await client.ping()
+            self._redis = client
+            self._script = client.register_script(_TOKEN_BUCKET_LUA)
+            log.info("llm.distributed: shared rate limiting active")
+        except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)